### PR TITLE
support resource action

### PR DESCRIPTION
**Pull request by adamshen**
1. filter params must be array, otherwise it can't be handled in server-side.
2. add a fetch_toke method 
3. support resource action

### CODE REVIEW

File: lib/manageiq/api/client/mixins/action_mixin.rb [hunk 1]
```diff
@@ -2,6 +2,16 @@ module ActionMixin
...
 
   private
+  def find_action(name)
+    @action = @actions.find { |action| action.name == name.to_s }
+  end
+
+  def do_action(name)
+    @action ||= find_action(name)
+    client.send @action.method, URI(@action.href) do
+      { action: @action.name }
```
Comment (abellotti) [inline lib/manageiq/api/client/mixins/action_mixin.rb]: needs to support

```ruby
{
     :action => action_name,
     :resource => {
        param1 => value1,
        param2 => value2,
        ...
     }
}
```

param1, param2, ... if any options are passed to the action (methd_missing args, &block).

File: lib/manageiq/api/client/mixins/action_mixin.rb [hunk 2]
```diff
@@ -2,6 +2,16 @@ module ActionMixin
...
+    end
```
Comment (abellotti) [inline lib/manageiq/api/client/mixins/action_mixin.rb]: should have an ActionResponse object, so callers can query res.success, res.message, etc.

File: lib/manageiq/api/client/connection.rb [hunk 3]
```diff
@@ -49,6 +49,12 @@ def json_response
           raise JSON::ParserError, "Response received from #{url} is not of type #{CONTENT_TYPE}"
         end
 
+        def fetch_token
+          @authentication.token = nil
+          res = get "auth"
+          @authentication.token = res["auth_token"]
+        end
```
Comment (abellotti) [inline lib/manageiq/api/client/connection.rb]: I take it this is a helper method so do an auth.  Would prefer do have an auth method on the client instance (miq), delegated to here is fine. i.e. miq.auth, returning and AuthResponse object, but have the caller specifically call an update_authentication with the new token,  miq.update_authentication could be called with user, password, token or group.

File: lib/manageiq/api/client/mixins/action_mixin.rb (hunk 2 above)
Comment (adamshen) [inline lib/manageiq/api/client/mixins/action_mixin.rb]: now it works like this

```ruby
vm = miq.vms.search.find {|vm| vm.data["name"]== 'ovirt'}
res = vm.start
res["success"] #=> true
```

Did you mean it should have a class named ActionResponse, then we could use its instance to store or show the result?

**Pull request closed without merging**